### ISSUE TITLE
Error in linking IFC file using relative path with saved blend file

### ISSUE THREAD

**Issue by maxfb87** [IfcOpenShell/IfcOpenShell#5768]
Try to link an external IFC file and check "use relative path", it raises this error
![image](https://github.com/user-attachments/assets/9c21fb45-564e-449f-bd33-a10be7372753)
but last week it worked

**Comment by BIMvoice** [comment]
I can add to this that linking has worked pretty badly lately. Many models don't get linked, yielding errors, and for those that do, even if they are small the linking happens very slowly. This used to work much smoother before, I don't know when or what happened. I'm using the daily release with Blender 4.2.3 LTS and I've been experiencing this for the past few weeks.

Same experience on Windows and Mac.

@Andrej730 any idea what happened?

**Comment by Andrej730** [comment]
> I can add to this that linking has worked pretty badly lately. Many models don't get linked, yielding errors, and for those that do, even if they are small the linking happens very slowly. This used to work much smoother before, I don't know when or what happened. I'm using the daily release with Blender 4.2.3 LTS and I've been experiencing this for the past few weeks.
> 
> Same experience on Windows and Mac.
> 
> @Andrej730 any idea what happened?

No idea, please, report the errors/ways to reproduce them.

**Comment by Andrej730** [comment]
> Try to link an external IFC file and check "use relative path", it raises this error

Can confirm, error does occur only if .blend file is saved and relative path is used. 
It also depends on how Blender ws started - if it was started as usually, then default cwd is `c:\users\xxx` leading to this error, if Blender was launched by opening .blend file, cwd changes to the .blend folder.

**Comment by Andrej730** [comment]
@maxfb87 should be fixed now, can you please test?

**Comment by maxfb87** [comment]
@Andrej730 yeah, now it works again, thanks!
Btw, regarding the load time, it takes about 20 sec to load a small ifc model and if i open the same ifc model in a separate session it is pretty instant...
So maybe it is possible to do some kind of optimization..

**Comment by Andrej730** [comment]
> Btw, regarding the load time, it takes about 20 sec to load a small ifc model and if i open the same ifc model in a separate session it is pretty instant...

Could it be that it's just the time it takes to start second instance Blender? (which happens when model is linked)

**Comment by maxfb87** [comment]
Yeah, it could be because it is similar to the time that blender takes to starts

**Comment by Andrej730** [comment]
Then I guess we're currently stuck with this delay since we do need a separate Blender session to create a .blend file to cache the linked model geometry.

**Comment by maxfb87** [comment]
Ok, just thinking loud, maybe it is possible to execute only necessary blender module to make the blender startup faster? For example sculptix and other modules aren't so much used i guess in this situation...

**Comment by Andrej730** [comment]
> Ok, just thinking loud, maybe it is possible to execute only necessary blender module to make the blender startup faster? For example sculptix and other modules aren't so much used i guess in this situation...

How? Blender core modules are not detachable.

In theory we can start blender as `blender --factory-startup` to avoid loading any addons but then we won't be able to access any bonsai methods / properties needed to load linked model. Therefore we'll need to either reimplement them in the linking script (which means maintaining same code in two places) or creating some kind of `bonsai.util` module that will be accessible from both Bonsai and linking script and will be limited not to use any Bonsai properties.

For me launching blender with `time blender --factory-startup --background` takes 0.4sec, launching `time blender --background` takes 3.9-5.0sec, so the difference is the time that will be possible to save loading Blender without addons.
